packages/valory/skills/abstract_round_abci/tests/test_tools/test_rounds.py

```diff
@@ -0,0 +1,678 @@
+        self.base_round_test = self.base_round_test_cls()
+        self.base_round_test._synchronized_data_class = (  # pylint: disable=protected-access
+            DummySynchronizedData
+        )
+        self.base_round_test.setup()
+        self.base_round_test._event_class = (  # pylint: disable=protected-access
+            DummyEvent
+        )
+        self.base_round_test.consensus_params._max_participants = (  # pylint: disable=protected-access
+            MAX_PARTICIPANTS
+        )
+
+    def create_test_gen(self, **kwargs: Any) -> None:
+        """Create the base test generator."""
+        test_method = getattr(self.base_round_test, self.test_method_name)
+        self.gen = test_method(**kwargs)
+
+    def exhaust_base_test_gen(self) -> None:
+        """Exhaust the base test generator."""
+        for _ in range(BASE_TEST_GEN_ITERATIONS):
+            next(self.gen)
+        last_iteration(self.gen)
+
+    def run_test(self, **kwargs: Any) -> None:
+        """Run a test for a base test."""
+        self.create_test_gen(**kwargs)
+        self.exhaust_base_test_gen()
+
+
+class DummyCollectDifferentUntilAllRoundWithEndBlock(
+    DummyCollectDifferentUntilAllRound
+):
+    """A `DummyCollectDifferentUntilAllRound` with `end_block` implemented."""
+
+    def __init__(self, dummy_exit_event: DummyEvent, *args: Any, **kwargs: Any):
+        """Initialize the dummy class."""
+        super().__init__(*args, **kwargs)
+        self.dummy_exit_event = dummy_exit_event
+
+    def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Enum]]:
+        """A dummy `end_block` implementation."""
+        if self.collection_threshold_reached and self.dummy_exit_event is not None:
+            return (
+                cast(
+                    DummySynchronizedData,
+                    self.synchronized_data.update(
+                        most_voted_keeper_address=list(self.collection.keys())
+                    ),
+                ),
+                self.dummy_exit_event,
+            )
+        return None
+
+
+class TestBaseCollectDifferentUntilAllRoundTest(BaseTestBase):
+    """Test `BaseCollectDifferentUntilAllRoundTest`."""
+
+    base_round_test: BaseCollectDifferentUntilAllRoundTest
+    base_round_test_cls = BaseCollectDifferentUntilAllRoundTest
+
+    @given(
+        st.one_of(st.none(), st.sampled_from(DummyEvent)),
+    )
+    def test_test_round(self, exit_event: DummyEvent) -> None:
+        """Test `_test_round`."""
+        self.base_round_test.consensus_params._max_participants = (  # pylint: disable=protected-access
+            MAX_PARTICIPANTS
+        )
+        test_round = DummyCollectDifferentUntilAllRoundWithEndBlock(
+            exit_event,
+            self.base_round_test.synchronized_data,
+            self.base_round_test.consensus_params,
+        )
+        round_payloads = [
+            (DummyTxPayload(f"agent_{i}", i)) for i in range(MAX_PARTICIPANTS)
```

Comment:
needless extra packing
`[DummyTxPayload(f"agent_{i}", i) for i in range(MAX_PARTICIPANTS)]`